grpc/runtime/src/main/scala/io/buoyant/grpc/runtime/GrpcStatus.scala

```diff
@@ -52,9 +74,21 @@ object GrpcStatus {
     case 16 => Unauthenticated(msg)
     case code => Other(code, msg)
   }
+  private[this] val StatusKey = "grpc-status"
+  private[this] val MessageKey = "grpc-message"
 
   def unapply(s: GrpcStatus): Option[(Int, String)] = Some((s.code, s.message))
 
+  def unapply(frame: h2.Frame): Option[GrpcStatus] =
```

Comment:
I'm on the fence about extractors like this.  Code like

```
frame match {
  case GrpcStatus(status) =>
  case _ =>
}
```

would be surprising to me because frame is not a GrpcStatus.  Or maybe more to the point, a method `GrpcStatus.apply(status: GrpcStatus): Frame` wouldn't make sense, so inverting it to an unapply maybe doesn't make sense either.